backend/api/badge_scanner/schema.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import pretix
+import re
+from typing import Any
+import strawberry
+from strawberry.types.info import Info
+from typing import Optional, Union
+
+from api.permissions import IsAuthenticated
+from badge_scanner import models
+from conferences.models import Conference
+from users.client import get_user_by_email
+
+
+@strawberry.type
+class ScanError:
+    message: str
+
+
+@strawberry.input
+class ScanBadgeInput:
+    url: str
+    conference_code: str
+
+    url_regex = re.compile(r"^https://pycon\.it/b/([\w\d-]+)$")
```

Comment:
we should move this outside so the RegEx will be compiled only once, instead of being compiled for every request (i.e every time ScanBadgeInput is created)

Reply:
it would look better outside regardless I think 🤔

Reply:
I prefer it being scoped to that class 😊

unless we move some of this logic somewhere else